community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactions.java

```diff
@@ -166,38 +143,16 @@ protected void dispose( KernelTransactionImplementation tx )
      */
     public List<KernelTransaction> activeTransactions()
     {
-        List<KernelTransaction> output = new ArrayList<>();
-        for ( KernelTransactionImplementation tx : allTransactions )
-        {
-            if ( tx.isOpen() )
-            {
-                output.add( tx );
-            }
-        }
-
-        return output;
+        return allTransactions.stream().filter( KernelTransaction::isOpen ).collect( toList() );
```

Comment:
Transactions are removed from the set when they are closed, so this filtering should not be necessary, right? If we returned a Set or a Collection, then we could wrap the `allTransactions` set in a read-only view. I think.